### PR TITLE
Add support for OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_EXPORTER_OTLP_HEADERS and OTEL_EXPORTER_OTLP_TIMEOUT env vars

### PR DESCRIPTION
**Pull request by dszmigielski**
Partially addresses [#1453](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1453).

This feature is important so that we can reuse the SDK in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation.

## Changes

The OtlpExporterOptions defaults can be overridden using `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_HEADERS` and `OTEL_EXPORTER_OTLP_TIMEOUT ` environmental variables as defined in the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md).

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs [hunk 1]
```diff
@@ -44,7 +97,7 @@ public class OtlpExporterOptions
         public int TimeoutMilliseconds { get; set; } = 10000;
 
         /// <summary>
-        /// Gets or sets the export processor type to be used with the OpenTelemetry Protocol Exporter. The default value is <see cref="ExportProcessorType.Batch"/>.
+        /// Gets or sets the export processor type to be used with the OpenTelemetry Protocol Exporter. The default value is <see cref="Batch{T}"/>.
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs:100]: Why make this change? I think the old version is correct.

Comment (dszmigielski) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs:100]: Seems like I got it done by some automatic refactoring. Thanks for catching that

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEventSource.cs [hunk 2]
```diff
@@ -81,5 +91,17 @@ public void CouldNotTranslateMetric(string className, string methodName)
         {
             this.WriteEvent(5, className, methodName);
         }
+
+        [Event(6, Message = "Failed to parse environment variable: '{0}', value: '{1}'.", Level = EventLevel.Warning)]
+        public void FailedToParseEnvironmentVariable(string name, string value)
+        {
+            this.WriteEvent(2, name, value);
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEventSource.cs:98]:
```suggestion
            this.WriteEvent(6, name, value);
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEventSource.cs [hunk 3]
```diff
@@ -81,5 +91,17 @@ public void CouldNotTranslateMetric(string className, string methodName)
...
+        }
+
+        [Event(7, Message = "Missing permissions to read environment variable: '{0}'", Level = EventLevel.Warning)]
+        public void MissingPermissionsToReadEnvironmentVariable(string exception)
+        {
+            this.WriteEvent(3, exception);
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OpenTelemetryProtocolExporterEventSource.cs:104]:
```suggestion
            this.WriteEvent(7, exception);
```

**Pull request merged**